test/README.md

```diff
@@ -0,0 +1,81 @@
+# WiredTiger Testing Overview
+
+## Introduction
+The reliability and robustness of WiredTiger is achieved in part by thorough and careful testing.
+
+- C tests: 26939 SLOC
+- CPP tests: 30 SLOC
+- Python tests: 83992 SLOC
+
+## Testing types
+There are many testing areas covered by WiredTiger's test infrasture. Some of the high level areas are:
+
+### Compatibility testing
+
+### Correctness testing
+- suite (Python / Unit tests)
+- csuite
+
+### Performance testing
+- wtperf
+- workgen
+
+### Sanitized testing
```

Comment:
I think there are two categories here. `Runtime correctness checking`. Examples are ASAN, MSAN, UBSAN, Valgrind (I'm not sure if we still use that, but I think we do).

`Static analysis`. Example here is Coverity, I think there are others.